homework7/Calculator/Calculator/CalculatorLogic.cs

```diff
@@ -0,0 +1,309 @@
+                OnPropertyChanged();
+            }
+        }
+
+        private string displayNumber = "0";
+        public string DisplayNumber
+        {
+            get
+            {
+                return displayNumber;
+            }
+            set
+            {
+                displayNumber = value;
+                OnPropertyChanged();
+            }
+        }
+
+        private string tempCalculationValue = "";
+        private char operationSign = ' ';
+        private char decimalPoint = ',';
+        private string errorMessage = "Error";
+
+        private enum State
+        {
+            NumberTyping,
+            NumberCopiedToDisplay,
+            EqualitySign,
+            AfterPoint,
+            Error,
+            Square
+        }
+
+        State currentState = State.NumberTyping;
+
+        public void Clear()
+        {
+            DisplayNumber = "0";
+            ExpressionOnTop = "";
+            tempCalculationValue = "";
+            operationSign = ' ';
+            currentState = State.NumberTyping;
+        }
+
+        public void PerformArithmeticOperationWithDisplayAndTempValue(char operation)
+        {
+            if (operationSign == ' ')
+            {
+                ExpressionOnTop = DisplayNumber + " " + operation.ToString();
+                tempCalculationValue = DisplayNumber;
+                operationSign = operation;
+                currentState = State.NumberCopiedToDisplay;
+            }
+            else
+            {
+                try
+                {
+                    var result = PerformArithmeticOperationWithStrings(tempCalculationValue, DisplayNumber, operationSign);
+                    ExpressionOnTop = result.ToString() + operation.ToString();
+                    tempCalculationValue = result.ToString();
+                    DisplayNumber = result.ToString();
+                    operationSign = operation;
+                    currentState = State.NumberCopiedToDisplay;
+
+                }
+                catch
+                {
+                    Clear();
+                    currentState = State.Error;
+                    DisplayNumber = errorMessage;
+                }
+            }
+        }
+
+        public void EqualitySign()
+        {
+            if (currentState != State.Error)
+            {
+                if (operationSign == ' ')
+                {
+                    ExpressionOnTop = DisplayNumber + " =";
+                    currentState = State.EqualitySign;
+                }
+                else
+                {
+                    try
+                    {
+                        if (currentState == State.EqualitySign)
+                        {
+                            var result = PerformArithmeticOperationWithStrings(DisplayNumber, tempCalculationValue, operationSign);
+                            ExpressionOnTop = DisplayNumber + " " + operationSign + " " + tempCalculationValue + " =";
```

Comment:
```suggestion
                            ExpressionOnTop = $"{DisplayNumber} {operationSign} {tempCalculationValue} =";
```
Если нужно конкатенировать несколько строк, используй форматную строку.
Так и читаемее, и эффективнее (несколько последовательных конкатенаций порождают промежуточные строки.
Может, компилятор это соптимизирует, а может и нет.